test/Microsoft.Extensions.Primitives.Tests/CompositeChangeTokenTest.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Moq;
+using Xunit;
+
+namespace Microsoft.Extensions.Primitives
+{
+    public class CompositeChangeTokenTest
+    {
+        [Fact]
+        public void RegisterChangeCallback_IsInvokedExactlyOnce()
+        {
+            // Arrange
+            var cancellationTokenSource = new CancellationTokenSource();
```

Comment:
Could we use more than one token source? Calling `Cancel` on the same token twice will no-op the second time.